src/app.rs

```diff
@@ -1,21 +1,52 @@
-use leptos::*;
-use serde_wasm_bindgen::from_value;
+use std::{collections::HashSet, net::IpAddr};
+
+use leptos::{html::Input, *};
+use serde::{Deserialize, Serialize};
+use serde_wasm_bindgen::{from_value, to_value};
 use wasm_bindgen::prelude::*;
 
 #[wasm_bindgen]
 extern "C" {
     #[wasm_bindgen(js_namespace = ["window", "__TAURI__", "tauri"])]
-    async fn invoke(cmd: &str) -> JsValue;
+    async fn invoke(cmd: &str, args: JsValue) -> JsValue;
 }
 
 type ServiceTypes = Vec<String>;
 
 async fn enum_service_types() -> ServiceTypes {
-    let service_types: ServiceTypes = from_value(invoke("enum_service_types").await).unwrap();
-    log::info!("Received {:#?}", service_types);
+    let service_types: ServiceTypes =
+        from_value(invoke("enum_service_types", JsValue::UNDEFINED).await).unwrap();
```

Comment:
Check whether this works:
```suggestion
        from_value(invoke("enum_service_types", ()).await).unwrap();
```

Reply:
#31